client/engine/chainservice/eth_chainservice.go

```diff
@@ -281,20 +287,60 @@ func (ecs *EthChainService) subscribeForLogs(ctx context.Context) {
 
 }
 
-// pollForLogs periodically polls the chain client to check if there new events.
-// It can function over a chain node that does not support notifications.
-func (ecs *EthChainService) pollForLogs(ctx context.Context) {
+// fetchLogsFromChain fetches logs from the chain from the given block number to the given block number.
+// It splits the query into multiple queries if the range is too large.
+func (ecs *EthChainService) fetchLogsFromChain(from *big.Int, to *big.Int) ([]ethTypes.Log, error) {
+
+	fromBlock := big.NewInt(0).Set(from)
+	toBlock := big.NewInt(0).Set(to)
+
+	logs := make([]ethTypes.Log, 0)
+
+	// Big.ints make it hard to parse but this is the condition:
+	// toBlock - fromBlock > MAX_QUERY_BLOCK_RANGE
+	for big.NewInt(0).Sub(toBlock, from).Cmp(big.NewInt(MAX_QUERY_BLOCK_RANGE)) > 0 {
+
+		nextBlock := big.NewInt(0).Add(fromBlock, big.NewInt(MAX_QUERY_BLOCK_RANGE))
+
+		query := ethereum.FilterQuery{
+			Addresses: []common.Address{ecs.naAddress},
+			FromBlock: fromBlock,
+			ToBlock:   toBlock,
+		}
+
+		fetchedLogs, err := ecs.chain.FilterLogs(context.Background(), query)
+
+		if err != nil {
+			return nil, err
+		}
+
+		logs = append(logs, fetchedLogs...)
+		// Update the fromBlock so it's the next block after the last block we fetched
+		fromBlock.Add(nextBlock, big.NewInt(1))
+	}
 
-	// TODO: We are currently querying from the genesis block to the current block.
-	// We could make this more performant by querying from the nitro adjudicator contract deployment block.
+	// This handles the final query which gets the remaining logs
 	query := ethereum.FilterQuery{
 		Addresses: []common.Address{ecs.naAddress},
-		FromBlock: nil,
-		ToBlock:   ecs.getCurrentBlockNum(),
+		FromBlock: fromBlock,
+		ToBlock:   toBlock,
 	}
 
 	fetchedLogs, err := ecs.chain.FilterLogs(context.Background(), query)
+	if err != nil {
+		return nil, err
+	}
+
+	logs = append(logs, fetchedLogs...)
+	return logs, nil
 
+}
+
+// pollForLogs periodically polls the chain client to check if there new events.
+// It can function over a chain node that does not support notifications.
+func (ecs *EthChainService) pollForLogs(ctx context.Context) {
+	toBlock := ecs.getCurrentBlockNum()
+	fetchedLogs, err := ecs.fetchLogsFromChain(big.NewInt(0), toBlock)
```

Comment:
Instead of starting at the genesis block we should use the block the contract was deployed. I'm just not sure where to find that information.

Reply:
I think that information is not readily available, unless we were to store it when we do the deploy https://github.com/ethers-io/ethers.js/discussions/2716#discussioncomment-2215706